mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -258,8 +272,14 @@ async fn process_passphrase_logic(
     match (wallet_name, passphrase) {
         (None, None) => Ok(None),
         // Legacy approach for passphrase, no `wallet_name` is needed in the config, in this case the passphrase is not encrypted and saved.
-        (None, Some(Passphrase::Decrypted(passphrase))) => Ok(Some(passphrase)),
         // Importing an encrypted passphrase without a wallet name is not supported since it's not possible to save the passphrase.
+        (None, Some(Passphrase::Decrypted(passphrase))) => {
+            if ctx.allow_registrations() {
+                Ok(Some(passphrase))
```

Comment:
I want to understand the need of `allow_registrations` here, this is the legacy approach where we pass a passphrase without a wallet name, so no wallet is created from KDF side here.

I understand the need for it in other places, although I think it's redundant, it can have benefits from GUI side for clear paths between registrations and login but GUI should never allow it's users to hit an error from KDF side such as `WalletInitError::WalletCreationNotAllowed` if I understand correctly. So, I don't think I also understand it's need in other places :)

I will take over this PR and it's fixes once I understand the need for this.

Reply:
@shamardy I'm fine leaving this part out because it won't be used in our GUI, but it will mean we have to entirely deprecate non-name wallet logins in the KDF SDK I'm working on. However, leaving this in will allow both register and sign-in for non-name wallets, giving devs more flexibility to use the SDK as they wish. From what I can tell, there is no downside for us.

Either way, I'll work with whatever your call is. It's worth noting that this slightly akin to the anonymous login offered by conventional auth services like Firebase Auth and Supabase Auth.

Reply:
> but it will mean we have to entirely deprecate non-name wallet logins in the KDF SDK I'm working on

non-name wallet logins still work the same way as before (before adding seed/mnemonic management in KDF) by just providing a passphrase.

Reply:
> > but it will mean we have to entirely deprecate non-name wallet logins in the KDF SDK I'm working on
> 
> non-name wallet logins still work the same way as before (before adding seed/mnemonic management in KDF) by just providing a passphrase with no `wallet_name` or `wallet_password`.

Understood. Currently, it's not possible to determine if a given seed has been used (registered) previously. It doesn't make a difference to KDF, but registration and sign-up are entirely different functions in the frontend SDK.

But yeah, we could leave this out since if users or developers need user management, they can migrate the seed to a named wallet.